### ISSUE TITLE
Restructure tests

### ISSUE THREAD

**Issue by mfocko** [packit-service/ogr#295]
Split tests into files like the implementations are.

As @lachmanfrantisek suggested, it would be great to have both old tests (checking that deprecated interfaces are still usable) and new ones, but this would lead to keeping duplicates of test_data, since they would check the same things, but each would need separate yaml for responses.

So I suggest transition from old to new tests right after 0.13.0 (or before 1.0.0, in case it will be released sooner)

**Comment by lachmanfrantisek** [comment]
@mfocko what do you think? Do we still want it.

**Comment by mfocko** [comment]
@lachmanfrantisek Yes, it's just waiting for the 0.13.0 release, since we've decided on keeping old tests to make sure the deprecated (but still present) functionality is working properly and restructuring those tests would mean keeping duplicate response files just because of different test names (and probably different call stack too).
